clippy_lints/src/methods/bind_instead_of_map.rs

```diff
@@ -104,6 +112,7 @@ pub(crate) trait BindInsteadOfMap {
                 let closure_args_snip = snippet(cx, closure_args_span, "..");
                 let option_snip = snippet(cx, args[0].span, "..");
                 let note = format!("{}.{}({} {})", option_snip, Self::GOOD_METHOD_NAME, closure_args_snip, some_inner_snip);
+                if !dry_run{
```

Comment:
Isn't it possible to return earlier, when `dry_run = true`? That way we don't need to run all of the code, without linting in the end.

Reply:
The return value is a boolean which means whether or not the lint was applicable to the input arguments. If we return earlier, we wouldn't be able to return a meaninful bool, would we?

The `bind-instead-of-map`'s dry-run is used in the `unnecessary-lazy-eval` lint. I will change that lint so that it calls the dry-run only if it itself is applicable and verifies there is not overlap with `bind-instead-of-map`.

Reply:
This sounds way too complicated/convoluted to me. Couldn't you do this easier, by just returning `true`/`false` in the `bind_instead_of_map` lints, when they trigger the lint and then handle this with a simple `if !linted { lint_unnecessary_lazy_eval }`? (See my comment below for more clarity)